core/src/main/scala/spinal/core/sim/package.scala

```diff
@@ -955,9 +970,16 @@ package object sim {
       } else {
         throw new Exception("???")
       }
-
     }
 
+    def forkStimulus() : Unit = {
+      val hz = cd.frequency match {
+        case ClockDomain.FixedFrequency(value) => value.toBigDecimal
```

Comment:
```suggestion
        case ClockDomain.FixedFrequency(value) => value.toBigDecimal,
        case _ => throw new Exception(s"Can't forkStimulus() w/o explicit frequency since frequency of ClockDomain $cd is not known")
```
?